core/execution/future/auction.go

```diff
@@ -61,6 +63,23 @@ func (m *Market) checkAuction(ctx context.Context, now time.Time, idgen common.I
 	}
 
 	if endTS := m.as.ExpiresAt(); endTS == nil || !endTS.Before(now) {
+		if isOpening && checkExceeded && m.as.ExceededMaxOpening(now) {
+			// cancel the market, exceeded opening auction
+			m.tradableInstrument.Instrument.Product.UnsubscribeTradingTerminated(ctx)
```

Comment:
don't we already have a close() that does all of these things correctly?

Reply:
You're right, I didn't use it because it takes `settlementDataInAsset *num.Uint` as an argument, but that isn't used if the market is being cancelled.